### PR TITLE
ALS-1669: Amf5beta/fix validators js

### PR DESCRIPTION
**Pull request by llibarona**

### CODE REVIEW

File: als-common/shared/src/main/scala/org/mulesoft/amfintegration/amfconfiguration/AmfConfigurationWrapper.scala [hunk 1]
```diff
@@ -211,3 +194,62 @@ object AmfConfigurationWrapper {
...
+                                        resourceLoaders: Seq[ResourceLoader]) {
+
+  def withValidators(v: Seq[AMFShapePayloadValidationPlugin]): AMFConfigurationStateManager =
+    AMFConfigurationStateManager(validators ++ v, syntaxes, dialects, resourceLoaders)
+
+  def withSyntaxes(s: Seq[AMFSyntaxParsePlugin]): AMFConfigurationStateManager =
+    AMFConfigurationStateManager(validators, syntaxes ++ s, dialects, resourceLoaders)
+
+  def withDialects(d: Seq[Dialect]): AMFConfigurationStateManager =
+    AMFConfigurationStateManager(validators, syntaxes, dialects ++ d, resourceLoaders)
```
Comment (agusayerza) [inline als-common/shared/src/main/scala/org/mulesoft/amfintegration/amfconfiguration/AmfConfigurationWrapper.scala:210]: Currently there is no way to remove/replace a dialect?

File: als-common/shared/src/main/scala/org/mulesoft/amfintegration/amfconfiguration/AmfConfigurationWrapper.scala [hunk 2]
```diff
@@ -32,21 +36,31 @@ import scala.concurrent.Future
...
+class AmfConfigurationWrapper private[amfintegration] (
+    private val initialConfig: AMFConfiguration,
+    val alsVocabularyRegistry: AlsVocabularyRegistry,
+    val alsDialectProvider: BaseAlsDialectProvider,
+    val resourceLoaders: Seq[ResourceLoader],
+    amfConfigurationStateDeltas: Option[AMFConfigurationStateManager] = None)
     extends PlatformSecrets {
   private implicit var configuration: AMFConfiguration = initialConfig
 
+  private var innerAmfConfigurationStateDeltas: AMFConfigurationStateManager =
```
Comment (agusayerza) [inline als-common/shared/src/main/scala/org/mulesoft/amfintegration/amfconfiguration/AmfConfigurationWrapper.scala:48]: Maybe a little picky on my part but I don't really like the name of the variable. I would remove the word "Deltas" and let it just be _innerAmfConfigurationState_ or _innerAmfConfigurationStateManager_

File: als-common/shared/src/main/scala/org/mulesoft/amfintegration/amfconfiguration/AmfConfigurationWrapper.scala (hunk 1 above)
Comment (llibarona) [reply als-common/shared/src/main/scala/org/mulesoft/amfintegration/amfconfiguration/AmfConfigurationWrapper.scala:210]: replacing should work by adding a dialect with the same ID or name/version, but I don't think removing dialects is part of the plan

Comment (agusayerza) [reply als-common/shared/src/main/scala/org/mulesoft/amfintegration/amfconfiguration/AmfConfigurationWrapper.scala:210]: LGTM then

**Review by agusayerza — APPROVED** [review]
LGTM

**Pull request merged**